Services/Certificate/classes/API/Repository/class.ilUserDataRepository.php

```diff
@@ -0,0 +1,197 @@
+<?php declare(strict_types=1);
+/* Copyright (c) 1998-2019 ILIAS open source, Extended GPL, see docs/LICENSE */
+namespace Certificate\API\Repository;
+
+use Certificate\API\Data\ilUserCertificateData;
+use Certificate\API\Filter\UserDataFilter;
+
+/**
+ * @author  Niels Theen <ntheen@databay.de>
+ * @ilCtrl_Calls: ilUserDataRepository: ilUserCertificateApiGUI
+ */
+class ilUserDataRepository
+{
+    /**
+     * @var \ilDBInterface
+     */
+    private $database;
+
+    /**
+     * @var \ilLogger
+     */
+    private $logger;
+
+    /**
+     * @var null|string
+     */
+    private $defaultTitle;
+
+    /**
+     * @var \ilCtrl
+     */
+    private $controller;
+
+    /**
+     * @param \ilDBInterface $database
+     * @param \ilLogger      $logger
+     * @param \ilCtrl        $controller
+     * @param string|null    $defaultTitle
+     */
+    public function __construct(
+        \ilDBInterface $database,
+        \ilLogger $logger,
+        \ilCtrl $controller,
+        string $defaultTitle = null
+    ) {
+        $this->database = $database;
+        $this->logger = $logger;
+        $this->controller = $controller;
+
+        if (null === $defaultTitle) {
+            global $DIC;
+            $defaultTitle = $DIC->language()->txt('certificate_no_object_title');
+        }
+        $this->defaultTitle = $defaultTitle;
+    }
+
+    /**
+     * @param UserDataFilter $filter
+     * @param array          $ilCtrlStack
+     * @return array
+     */
+    public function getUserData(UserDataFilter $filter, array $ilCtrlStack) : array
+    {
+        $userIds = $filter->getUserIds();
+
+        $sql = '
+SELECT 
+  il_cert_user_cert.pattern_certificate_id,
+  il_cert_user_cert.obj_id,
+  il_cert_user_cert.user_id,
+  il_cert_user_cert.user_name,
+  il_cert_user_cert.acquired_timestamp,
+  il_cert_user_cert.currently_active,
+  il_cert_user_cert.id,
+  (CASE WHEN (object_data.title IS NULL)
```

Comment:
I don't think this has any significant negative impact on the performance. If this would be used for a `JOIN` this might be an issue. If you detect and performance issues with this, a Mantis ticket would be appreciated :-).